fdb-relational-core/src/test/java/com/apple/foundationdb/relational/recordlayer/RelationalArrayTest.java

```diff
@@ -124,18 +125,14 @@ private void insertArraysViaQuerySimpleStatement() throws SQLException {
                         "[x'31', x'32'], null, " +
                         "[(11, '11'), (22, '22')], null " +
                         ")"));
-        // This should not go through since non-null array is not initialized, However, that is being currently getting
-        // initialized by an empty array (no elements in unwrapped REPEATED field in proto).
-        // This phenomenon is also true with basic types, i.e., non-nullability is not enforced and is tracked by
-        // TODO (Add support + tests for column nullable/not null)
-        insertQuery("INSERT INTO T (pk) VALUES (4)");
+        Assert.assertThrows(SQLException.class, () -> insertQuery("INSERT INTO T (pk) VALUES (4)"));
```

Comment:
Can you use `RelationalAssertions.assertThrowsSqlException` that allows you to check for the error message and SQLState?

Reply:
done!

Reply:
Can you check the SQLState as well? 
`RelationalAssertions.assertThrowsSqlException` has a method `withErrorCode` or something like this that you can use.